VerilogCodingStyle.md

```diff
@@ -2539,6 +2539,52 @@ function automatic logic [2:0] foo(logic [2:0] a, logic [2:0] b);
 endfunction
 ```
 
+Functions should not reference any non-local signals, variables, or memories
+outside their scope. Avoiding non-local references improves readability and
+helps reduce simulation/synthesis mismatches. Accessing non-local parameters
```

Comment:
Maybe add "and constants" because they are of similar nature (as in, would not change at runtime).